File: woodstorage/src/main/java/com/jordylangen/woodstorage/WoodStorageTree.java
package com.jordylangen.woodstorage;

import androidx.annotation.NonNull;
import io.reactivex.processors.PublishProcessor;
import timber.log.Timber;

public class WoodStorageTree extends Timber.DebugTree {

    private PublishProcessor<LogEntry> logEntryPublishProcessor;

    WoodStorageTree(PublishProcessor<LogEntry> logStatementPublishSubject) {
        this.logEntryPublishProcessor = logStatementPublishSubject;
    }

    @Override
    protected void log(int priority, String tag, @NonNull String message, Throwable t) {
        LogEntry logEntry = new LogEntry(tag, priority, message);
        logEntryPublishProcessor.onNext(logEntry);
    }
}

File: woodstorage/src/main/java/com/jordylangen/woodstorage/WoodStorageTree.kt
package com.jordylangen.woodstorage

import io.reactivex.processors.PublishProcessor
import timber.log.Timber

class WoodStorageTree internal constructor(private val logEntryPublishProcessor: PublishProcessor<LogEntry>) :
    Timber.DebugTree() {

    override fun log(priority: Int, tag: String?, message: String, t: Throwable?) {
        val logEntry = LogEntry(tag!!, priority, message)
        logEntryPublishProcessor.onNext(logEntry)
    }
}
